src/IO/ADIOS/ADIOS2IOHandler.cpp

```diff
@@ -1244,20 +1364,61 @@ namespace detail
         }
         if( m_engine )
         {
+            if( streamStatus == StreamStatus::DuringStep )
+            {
+                m_engine.get().EndStep();
+            }
             m_engine.get().Close();
+            m_ADIOS.RemoveIO( m_IOName );
         }
     }
 
-    void
-    BufferedActions::configure_IO( ADIOS2IOHandlerImpl & impl )
-    {
-        (void)impl;
+    void BufferedActions::configure_IO(ADIOS2IOHandlerImpl& impl){
+        ( void )impl;
+        static std::set< std::string > streamingEngines = {
+            "sst", "insitumpi", "inline", "staging"
+        };
+        static std::set< std::string > fileEngines = {
+            "bp4", "bp3", "hdf5", "file"
+        };
+
+        // set engine type
+        {
+            // allow overriding through environment variable
```

Comment:
Is the logic here: json option overwrites env var?

Reply:
Other way around actually. By default, pick the one given in the JSON string (that one's written to `m_engineType` earlier in `ADIOS2IOHandlerImpl::init`. Environment variable allows overwriting.